### PR TITLE
misc: Fix bash path in shebangs

### PR DESCRIPTION
**Pull request by asomers**
/bin/bash is a Linuxism.  Other operating systems install bash to
different paths.  Use /usr/bin/env in shebangs to find bash.

### CODE REVIEW

**Comment by asomers** [comment]
CC @wjw , who's done a lot of work on Ceph portability

File: qa/workunits/cephtool/test.sh [hunk 1]
```diff
@@ -1,10 +1,10 @@
-#!/bin/bash -x
+#!/usr/bin/env bash
 # -*- mode:shell-script; tab-width:8; sh-basic-offset:2; indent-tabs-mode:t -*-
 # vim: ts=8 sw=8 ft=bash smarttab
 
 source $(dirname $0)/../ceph-helpers.sh
 
-set -e
+set -ex
```
Comment (wjwithagen) [inline qa/workunits/cephtool/test.sh]: @asomers 
This change silences the -x on the ceph-helpers.sh, not sure you want to change behaviour in such an invasive patch.

File: src/test/encoding/check-generated.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -e
+#!/usr/bin/env bash
```
Comment (wjwithagen) [inline src/test/encoding/check-generated.sh]: @asomers 
Missing `-e`

File: src/test/encoding/readable.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -e
+#!/usr/bin/env bash
```
Comment (wjwithagen) [inline src/test/encoding/readable.sh]: @asomers 
Missing -e

**Comment by wjwithagen** [comment]
On 22-7-2017 01:01, Alan Somers wrote:
> CC @wjw <https://github.com/wjw> , who's done a lot of work on Ceph
> portability

Hi Alan,

Welcome to ceph-dev, lets see if we can get BlueStore into FreeBSD...
Big patch to kick off with, but lets see how we fare.

Could you amend your PR text with:
    prefix the title with a category of what your change.
	I would probably here use: "scripts:"
    Copy the text in commit also in the text of the PR.

You already managed to get a merge conflict in:
    qa/workunits/mon/workloadgen.sh

Which is going to happen more often because of the number of files
modified. Normally I chose to make smaller PRs because they tend to get
accepted earlier. Large PRs just take a long time because they also need
to run thru QA and bigger patches run larger chance of errors or conflicts,

Not sure if you already found:
	./SubmittingPatches.rst

--WjW

File: src/test/encoding/check-generated.sh (hunk 2 above)
Comment (asomers) [inline src/test/encoding/check-generated.sh]: Actually, not.  There's already a redundant "set -e" on line 8

File: src/test/encoding/readable.sh (hunk 3 above)
Comment (asomers) [inline src/test/encoding/readable.sh]: Likewise, there's already a redundant `set -e` on line 9

**Comment by asomers** [comment]
It turns out that some submodules also use /bin/bash shebangs.  So I'm reverting the changes I made to README.FreeBSD.  But the other changes are still good.

**Comment by wjwithagen** [comment]
jenkins retest please

**Comment by wjwithagen** [comment]
@asomers 
Hi Alan,
You need to remove ceph-disk from the commit, since the code that had a bash ref in it has gone.

File: qa/standalone/scrub/osd-scrub-repair.sh [hunk 4]
```diff
@@ -14,7 +14,8 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU Library Public License for more details.
 #
-source $CEPH_ROOT/qa/standalone/ceph-helpers.sh
+set -x
+source $CEPH_ROOT/qa/workunits/ceph-helpers.sh
```
Comment (wjwithagen) [inline qa/standalone/scrub/osd-scrub-repair.sh]: @asomers 
You are moving code back to workunits, were a few commits back things got moved to standalone.

File: qa/workunits/fs/misc/layout_vxattrs.sh [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -x
+#!/usr/bin/env bash
 
 set -e
 set -x
```
Comment (wjwithagen) [inline qa/workunits/fs/misc/layout_vxattrs.sh]: @asomers 
Why not join them on one line?

File: qa/workunits/fs/quota/quota.sh [hunk 6]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 set -e
 set -x
```
Comment (wjwithagen) [inline qa/workunits/fs/quota/quota.sh]: @asomers Dito, unless you only want to change what really needs to change.

**Comment by wjwithagen** [comment]
@tchaikov 
Could you take a look at this as well, and run it thru QA?

File: qa/standalone/scrub/osd-scrub-repair.sh (hunk 4 above)
Comment (asomers) [inline qa/standalone/scrub/osd-scrub-repair.sh]: grahh, missed one.  I'll fix it.

File: qa/workunits/fs/misc/layout_vxattrs.sh (hunk 5 above)
Comment (asomers) [inline qa/workunits/fs/misc/layout_vxattrs.sh]: Sure, I could do that too.

**Comment by wjwithagen** [comment]
Jenkins retest

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-08-16_17:28:59-rados-wip-kefu-testing-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-08-17_11:03:01-rados-wip-kefu-testing-distro-basic-mira/

File: systemd/ceph [hunk 7]
```diff
@@ -1,4 +1,4 @@
-#! /bin/bash
+#! /usr/bin/env bash
```
Comment (tchaikov) [inline systemd/ceph]: i think systemd is used by Linux only. so this change is not necessary?

Comment (wjwithagen) [inline systemd/ceph]: @tchaikov 
I happy to say it is Linux only...
The amount of unexpected irritation I hear from my colleague makes me a content init user.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-08-20_01:49:41-rados-wip-kefu-testing-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-08-19_02:32:27-rados-wip-kefu-testing-distro-basic-mira/

**Comment by liewegas** [comment]
@tchaikov ok to merge?

**Pull request merged**